robotbase/src/main/java/frc/robot/commands/pivot/PivotZero.java

```diff
@@ -1,33 +1,43 @@
 package frc.robot.commands.pivot;
 
-import edu.wpi.first.wpilibj.DriverStation;
 import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.ParallelDeadlineGroup;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import edu.wpi.first.wpilibj2.command.WaitCommand;
 import frc.robot.Constants;
 import frc.robot.Robot;
+import frc.robot.controls.util.AxisInterface;
 
-public class PivotZero extends Command {
-  public boolean isZeroed() {
-    return Robot.pivot.getPivotVelocity() <= Constants.SHOOTER.ZERO_SPEED_STOP_TOLERANCE;
-  }
-
-  @Override
-  public void initialize() {
-    Robot.pivot.setPivotAxisSpeed(Constants.SHOOTER.ZERO_SPEED);
-  }
-
-  @Override
-  public boolean isFinished() {
-    return isZeroed();
+public class PivotZero extends ParallelDeadlineGroup {
+  public static class UntilPivotZero extends Command {
+    @Override
+    public boolean isFinished() {
+      return Math.abs(Robot.pivot.getVelocity()) < Constants.PIVOT.ZERO_SPEED_STOP_TOLERANCE;
+    }
   }
 
-  @Override
-  public void end(boolean interrupted) {
-    Robot.pivot.stop();
+  public PivotZero() {
+    super(
+        // First wait until we start moving, then wait until the velocity stops
+        new SequentialCommandGroup(
+                new WaitCommand(Constants.PIVOT.ZERO_SPEED_INITIAL_SECONDS), new UntilPivotZero())
+            .finallyDo(
+                (boolean interrupted) -> {
+                  if (!interrupted) {
+                    Robot.pivot.setZero();
+                  } else {
+                    System.err.println("Pivot Zero interrupted!");
```

Comment:
Would it be better to do a DriverStation.reportError here?

Reply:
I originally had that, but I don't like that it takes extra lines to display the line of code on which the error occurred.